com.hazelcast.jet.impl.connector.ReadMapOrCacheP_ConsistencyTest.test_removingItems_remote

http://jenkins.hazelcast.com/job/jet-oss-pr-builder/2028/testReport/junit/com.hazelcast.jet.impl.connector/ReadMapOrCacheP_ConsistencyTest/test_removingItems_remote/

```
Error Message
com.hazelcast.jet.JetException: java.lang.IllegalStateException: Unable to connect to any cluster.
Stacktrace
java.util.concurrent.CompletionException: com.hazelcast.jet.JetException: java.lang.IllegalStateException: Unable to connect to any cluster.
	at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1934)
	at com.hazelcast.jet.Job.join(Job.java:137)
	at com.hazelcast.jet.impl.connector.ReadMapOrCacheP_ConsistencyTest.test_removingItems(ReadMapOrCacheP_ConsistencyTest.java:171)
	at com.hazelcast.jet.impl.connector.ReadMapOrCacheP_ConsistencyTest.test_removingItems_remote(ReadMapOrCacheP_ConsistencyTest.java:111)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.jet.JetException: java.lang.IllegalStateException: Unable to connect to any cluster.
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$null$8(JobCoordinationService.java:320)
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:822)
	at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:797)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:59)
	at com.hazelcast.jet.impl.MasterJobContext.setFinalResult(MasterJobContext.java:492)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$null$19(MasterJobContext.java:673)
	at com.hazelcast.jet.impl.util.ExceptionUtil.lambda$withTryCatch$0(ExceptionUtil.java:180)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760)
	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:778)
	at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2140)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$finalizeJob$20(MasterJobContext.java:669)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:978)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$43(JobCoordinationService.java:997)
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
Caused by: java.lang.IllegalStateException: Unable to connect to any cluster.
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.doConnectToCluster(ClientConnectionManagerImpl.java:397)
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.connectToCluster(ClientConnectionManagerImpl.java:346)
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.start(ClientConnectionManagerImpl.java:296)
	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.start(HazelcastClientInstanceImpl.java:365)
	at com.hazelcast.client.HazelcastClient.constructHazelcastClient(HazelcastClient.java:460)
	at com.hazelcast.client.HazelcastClient.newHazelcastClientInternal(HazelcastClient.java:416)
	at com.hazelcast.client.HazelcastClient.newHazelcastClient(HazelcastClient.java:136)
	at com.hazelcast.jet.impl.connector.ReadMapOrCacheP$RemoteProcessorSupplier.init(ReadMapOrCacheP.java:301)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initProcSuppliers(ExecutionPlan.java:305)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:146)
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:134)
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$808bb183$1(JobExecutionService.java:232)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:30)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:408)
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:232)
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:73)
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:228)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:217)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:405)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:432)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:590)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:575)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:534)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:236)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59)
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:261)
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:245)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:227)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:978)
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:987)
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:977)
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:186)
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:887)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:246)
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/MulticastDiscoveryTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/MulticastDiscoveryTest.java
@@ -24,17 +24,20 @@ import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.jet.config.JetClientConfig;
 import com.hazelcast.jet.core.JetTestSupport;
 import com.hazelcast.jet.core.TestProcessors;
+import com.hazelcast.jet.test.SerialTest;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
 
 import static org.junit.Assert.assertEquals;
 
 @RunWith(HazelcastSerialClassRunner.class)
+@Category(SerialTest.class)
 public class MulticastDiscoveryTest extends JetTestSupport {
 
     private static final String UNABLE_TO_CONNECT_MESSAGE = "Unable to connect";

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/HazelcastRemoteConnectorTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/HazelcastRemoteConnectorTest.java
@@ -36,6 +36,7 @@ import com.hazelcast.jet.core.DAG;
 import com.hazelcast.jet.core.JetTestSupport;
 import com.hazelcast.jet.core.Vertex;
 import com.hazelcast.jet.core.processor.SourceProcessors;
+import com.hazelcast.jet.test.SerialTest;
 import com.hazelcast.map.EventJournalMapEvent;
 import com.hazelcast.projection.Projections;
 import com.hazelcast.query.Predicates;
@@ -44,6 +45,7 @@ import org.junit.AfterClass;
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
 import java.util.Map;
@@ -69,6 +71,7 @@ import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
 @RunWith(HazelcastSerialClassRunner.class)
+@Category(SerialTest.class)
 public class HazelcastRemoteConnectorTest extends JetTestSupport {
 
     private static final int ITEM_COUNT = 20;

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/ReadMapOrCacheP_ConsistencyTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/ReadMapOrCacheP_ConsistencyTest.java
@@ -28,11 +28,13 @@ import com.hazelcast.jet.core.JetTestSupport;
 import com.hazelcast.jet.pipeline.BatchSource;
 import com.hazelcast.jet.pipeline.Pipeline;
 import com.hazelcast.jet.pipeline.test.AssertionSinks;
+import com.hazelcast.jet.test.SerialTest;
 import com.hazelcast.map.IMap;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
 import java.util.ArrayList;
@@ -53,6 +55,7 @@ import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
 @RunWith(HazelcastSerialClassRunner.class)
+@Category(SerialTest.class)
 public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
 
     private static final String MAP_NAME = "map";

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/test/SerialTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/test/SerialTest.java
@@ -18,6 +18,9 @@ package com.hazelcast.jet.test;
 
 /**
  * Category marker for the tests which are executed in series within a single JVM
+ * <p>
+ * These typically need to create a real Hazelcast or Jet instance which can
+ * conflict with other such instances created in other tests
  */
 public interface SerialTest {
 }

--- a/hazelcast-jet-distribution/src/root/README.md
+++ b/hazelcast-jet-distribution/src/root/README.md
@@ -22,12 +22,8 @@ What's Included
 * `config/hazelcast-client.yaml`: The client configuration file used by the 
   Jet Command Line client
 * `config/log4j.properties`: Logging configuration used by the Jet Instance
-* `opt`: Optional plugins, sources and sinks for Jet. The following are included:
-    * Kafka
-    * Hadoop
-    * S3 
-    * Avro
-    * Python
+* `opt`: Optional extensions for Jet. You can include them in the classpath by moving
+them to the lib folder.
    
 Quickstart
 ----------

